lib/compiler.js

```diff
@@ -538,3 +525,61 @@ function resolveRelativePaths(relativePaths, appRootDir) {
     }
   });
 }
+
+function isPreferredExtension (filename, disallowedExtensions) {
+  disallowedExtensions = disallowedExtensions || [];
+
+  var includeExtensions = require.extensions;
+  var excludeExtensions = {};
+  excludeExtensions[FILE_EXTENSION_JSON] = FILE_EXTENSION_JSON;
+
+  disallowedExtensions.forEach(function(ext) {
+    excludeExtensions[ext] = ext;
+  });
+
+  var ext = path.extname(filename);
+  return (ext in includeExtensions) && !(ext in excludeExtensions);
+}
+
+function fixFileExtension (filepath, files, disallowedExtensions,
+  contentDataType) {
```

Comment:
Will it be a good idea, to combine `disallowedExtensions` and `contentDataType` parameters, into a single parameter like `fixOptions`. An example for `fixOptions` would be
```
{
  'disallowedExtensions': ['.node'],
  'contentDataType': 'functions' 
}
```